src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/SortKey.Linux.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Runtime.Versioning;
+
+namespace System.DirectoryServices.Protocols
+{
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
```

Comment:
The split nature of this is going to cause complications in the reference assembly generation, not to mention that I genuinely don't know how the Marshaller handles reference types (other than a special list of them).

Any P/Invokes that are using this type should change to using a non-public struct that receives a copy of the data in here.  And that structure can certainly vary by platform.

Reply:
Thanks for the suggestion! I updated the PR by introducing a new `SortKeyInterop` internal class that is used for the P/Invoke calls.